simulation/complex_networks/complex_random_topology_generator.py

```diff
@@ -10,91 +10,103 @@
 import sys
 import networkx as nx
 import numpy as np
-#make a class this module
+# make a class this module
+
+
 def graph_to_str(adj_matrix):
     str_matrix = []
     v = int(math.sqrt(len(adj_matrix)))
-    for i in range (1, v):
-        for j in range(i+1, v+1):
-            index = ( i - 1 ) * v + j - 1
-            if adj_matrix[ index ]:
+    for i in range(1, v):
+        for j in range(i + 1, v + 1):
+            index = (i - 1) * v + j - 1
+            if adj_matrix[index]:
                 str_matrix.append(str(i) + " " + str(j))
     return str_matrix
 
 # Return a random integer between 0 and k-1 inclusive.
-def ran( k ):
-    return random.randint(0, k-1)
+
+
+def ran(k):
+    return random.randint(0, k - 1)
+
 
 def nx_graph_to_adj_matrix(G):
     nmp_matrix = nx.to_numpy_matrix(G, dtype=int)
     fl = nmp_matrix.flatten()
     adj_matrix = np.squeeze(np.asarray(fl))
     return adj_matrix
 
+
 def random_connected_graph(v, e):
     adj_matrix = [0] * v * v
     tree = [0] * v
     init_array(tree, v)
     tree = permute(tree)
     for i in range(1, v):
-        j = ran( i )
-        adj_matrix[ tree[ i ] * v + tree[ j ] ] = 1
-        adj_matrix[ tree[ j ] * v + tree[ i ] ] = 1
+        j = ran(i)
+        adj_matrix[tree[i] * v + tree[j]] = 1
+        adj_matrix[tree[j] * v + tree[i]] = 1
     count = v - 1
     while count < e:
-        i = ran( v )
-        j = ran( v )
+        i = ran(v)
+        j = ran(v)
         if i == j:
             continue
-        if i > j :
+        if i > j:
             i, j = j, i
         index = i * v + j
-        if not adj_matrix[ index ]:
-            adj_matrix[ index ] = 1
+        if not adj_matrix[index]:
+            adj_matrix[index] = 1
             count += 1
     return adj_matrix
 
+
```

Comment:
why one more line?

Reply:
According to pep8 standarts there has to be 2 newlines before function definitions so that it’s clear that function is separated from others.